### PR TITLE
Add edge behaviour

### PR DESCRIPTION
**Pull request by q349wang**
Add edge behaviour for gfx that overflows screen

### CODE REVIEW

File: gfx/include/gfx/gfx_constants.h [hunk 1]
```diff
@@ -6,10 +6,10 @@
 
 namespace antioch::gfx {
 
-constexpr uint32_t MAX_PRIMS_PER_DRAW = 50;
-constexpr uint32_t MAX_DRAWS_PER_BUFFER = 1;
+constexpr uint32_t MAX_PRIMS_PER_DRAW = 64;
```
Comment (Victor-C-Zhang) [inline gfx/include/gfx/gfx_constants.h:9]: what's a prim

File: gfx/src/base/device.cpp [hunk 2]
```diff
@@ -176,33 +182,48 @@ GFX_API Result Device::submit(uint32_t submitCount, const SubmitInfo* pSubmits)
...
+                      screenPos.y < 0) {
+                    continue;
+                  }
+                  break;
+                }
+                case EdgeBehaviour::eWrap: {
+                  if (screenPos.x >= width || screenPos.y >= height || screenPos.x < 0 ||
+                      screenPos.y < 0) {
+                    if (screenPos.x >= width || screenPos.x < 0) {
+                      screenPos.x %= width;
```
Comment (Victor-C-Zhang) [inline gfx/src/base/device.cpp:207]: can you get away with just a -=?

**Review by Victor-C-Zhang — APPROVED** [review]

File: gfx/src/base/device.cpp (hunk 2 above)
Comment (q349wang) [reply gfx/src/base/device.cpp:207]: not if the x position is more than double the width

File: gfx/include/gfx/gfx_constants.h (hunk 1 above)
Comment (q349wang) [reply gfx/include/gfx/gfx_constants.h:9]: Primitive shape, basically the objects that get drawn

File: gfx/src/base/device.cpp (hunk 2 above)
Comment (Victor-C-Zhang) [reply gfx/src/base/device.cpp:207]: when will this happen? When will we want to draw a shape that's both (a) larger than the width and (b) offset more than 100% of width?

**Pull request merged**